clients/snowflake/snowflake.go

```diff
@@ -62,14 +64,16 @@ func (s *Store) GetConfigMap() *types.DwhToTablesConfigMap {
 }
 
 func (s *Store) Merge(tableData *optimization.TableData) error {
-	// TODO: Implement max retry count
-	err := shared.Merge(s, tableData, s.config, types.MergeOpts{})
-	if IsAuthExpiredError(err) {
-		slog.Warn("Authentication has expired, will reload the Snowflake store and retry merging", slog.Any("err", err))
-		s.reestablishConnection()
-		return s.Merge(tableData)
+	var err error
+	for i := 0; i < maxRetries; i++ {
+		err = shared.Merge(s, tableData, s.config, types.MergeOpts{})
+		if IsAuthExpiredError(err) {
+			slog.Warn("Authentication has expired, will reload the Snowflake store and retry merging", slog.Any("err", err))
```

Comment:
Would be slightly better to reestablish the connection before the merge when `i > 0`.